packages/core/src/shared/featureConfig.ts

```diff
@@ -32,8 +32,11 @@ const customizationArnOverrideName = 'customizationArnOverride'
 const featureConfigPollIntervalInMs = 30 * 60 * 1000 // 30 mins
 const dataCollectionFeatureName = 'IDEProjectContextDataCollection'
 
+const featureNames = [testFeatureName, customizationArnOverrideName, dataCollectionFeatureName] as const
+export type FeatureName = (typeof featureNames)[number]
```

Comment:
You can avoid double defining featureNames by defining them like this instead:
```
const Features = {
    test: 'testFeatureName',
    customizationArnOverrideName: 'customizationArnOverrideName',
    dataCollectionFeatureName: 'IDEProjectContextDataCollection'
} as const

type FeatureName = (typeof features)[keyof typeof features]
```

Then ofc instead of `dataCollectionFeatureName` variable do `Features.dataCollectionFeatureName`